app/src/main/java/com/example/paulsuarez/a28_filterable_lists/models/Constructor.java

```diff
@@ -0,0 +1,29 @@
+package com.example.paulsuarez.a28_filterable_lists.models;
+
+import android.content.Intent;
+
+public class Constructor {
+    public String name;
+    public String rank;
+    public String bio;
+
+    public Constructor(String name, String rank, String bio) {
+        this.name = name;
+        this.rank = rank;
+        this.bio = bio;
+    }
+
+    public void fillIntent(Intent intent) {
```

Comment:
This is a perfectly reasonable approach in this case since the data is small. The other approach that is often taken, especially when your model starts to get bigger, is to pass some sort of id in the intent rather than all the data, and have the receiver just look up the correct record. No need to change this, just something to keep in mind as an option.